frontend-api/src/main/java/uk/gov/di/authentication/frontendapi/lambda/VerifyMfaCodeHandler.java

```diff
@@ -252,7 +253,15 @@ private APIGatewayProxyResponseEvent verifyCode(
         sessionService.storeOrUpdateSession(session);
 
         return errorResponseMaybe
-                .map(response -> generateApiGatewayProxyErrorResponse(400, response))
+                .map(
+                        response -> {
```

Comment:
So, if possible and if it doesn't end up being a rabbit hole, it'd be good to generate this response where the other error responses are generated, so that we don't end up with this specific reauth logic here. Same with where the counts get incremented. 

This should hopefully mean that if we get to refactoring this stuff to put things in more sensible places, all the count and error logic is in one place.

If it ends up being a rabbit hole, happy to approve as is on the basis that we're going to think about how to refactor the mfa stuff.

Reply:
Let's defer this until the refactor.